File: app/src/main/java/com/example/hw3_b/MusicAdapter.java
package com.example.hw3_b;

import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import java.util.List;

public class MusicAdapter extends RecyclerView.Adapter<MusicAdapter.MusicViewHolder> {

    public List<SongOls> getList() {
        return list;
    }

    List<SongOls> list;
    OnItemClickListener onItemClickListener;

    public MusicAdapter(List<SongOls> list, OnItemClickListener onItemClickListener) {
        this.list = list;
        this.onItemClickListener = onItemClickListener;
    }

    @Override
    public MusicViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.item_view, parent, false);
        return new MusicViewHolder(view);
    }

    @Override
    public void onBindViewHolder(MusicViewHolder holder, int position) {
        holder.bind(list.get(position));
    }

    @Override
    public int getItemCount() {
        return list.size();
    }

    class MusicViewHolder extends RecyclerView.ViewHolder {
        TextView tv_name;
        public MusicViewHolder(View view) {
            super(view);
            tv_name = view.findViewById(R.id.tv_nameOfSong);
            view.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View view) {
                    SongOls song = list.get(getLayoutPosition());
                    onItemClickListener.onItemClick(song);
                }
            });
        }

        public void bind(SongOls song) {
            tv_name.setText(song.getName());
        }
    }

        public interface OnItemClickListener{
            void onItemClick(SongOls song);
        }
}

File: app/src/main/java/com/example/musicplayerkt/MusicAdapter.kt
package com.example.musicplayerkt

import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
class MusicAdapter(private val list: List<Song>, private val onItemClickListener: OnItemClickListener) :
    RecyclerView.Adapter<MusicAdapter.MusicViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MusicViewHolder {
        val view = LayoutInflater.from(parent.context)
            .inflate(R.layout.item_view, parent, false)
        return MusicViewHolder(view)
    }

    override fun onBindViewHolder(holder: MusicViewHolder, position: Int) {
        holder.bind(list[position])
    }

    override fun getItemCount(): Int {
        return list.size
    }

    inner class MusicViewHolder(val view: View) : RecyclerView.ViewHolder(view) {
        fun bind(song: Song) = with(itemView) {
            tv_name.text = song.name
            view.setOnClickListener {
                val song = list[layoutPosition]
                onItemClickListener.onItemClick(song)
            }
        }
    }

    interface OnItemClickListener {
        fun onItemClick(song: Song)
    }
}
